### PR TITLE
Allocation priority field

### LINKED ISSUE #273: Flight allocationPriority field
Adds the `allocationPriority` int from PRX/augury.prx.org#347 to the flight form.

image

- [ ] Just add a numeric field.  Can be negative.  Default it to 10.

### PR DESCRIPTION
**Pull request by cavis**
Closes #273.

- [x] Adds the `allocationPriority` numeric field
- [x] Also throws a 500ms debounce on inventory preview.  (Those number fields fire off a ton of preview requests if you use the up/down arrows).

### CODE REVIEW

File: src/app/campaign/flight/flight-form.component.html [hunk 1]
```diff
@@ -47,7 +47,7 @@
     <mat-form-field class="flight-form-field" appearance="outline">
       <mat-label>Series</mat-label>
       <mat-select formControlName="set_inventory_uri" required>
-        <mat-option *ngFor="let inv of inventory" [value]="inv.self_uri">{{ inv.podcastTitle }}</mat-option>
+        <mat-option *ngFor="let inv of inventory" [value]="inv.self_uri">{{ inv.podcastId }}: {{ inv.podcastTitle }}</mat-option>
```
Comment (cavis) [inline src/app/campaign/flight/flight-form.component.html:50]: We display podcast ids elsewhere, so why not here too?

File: src/app/campaign/store/actions/campaign-action.service.ts [hunk 2]
```diff
@@ -33,6 +33,8 @@ export class CampaignActionService implements OnDestroy {
     withLatestFrom(this.store.pipe(select(selectRoutedFlight))),
     // re-using the flight form, so check that ids match
     filter(([formState, flightState]) => !flightState || !flightState.localFlight || formState.flight.id === flightState.localFlight.id),
+    // debounce so numeric fields aren't constantly firing previews
+    debounceTime(500),
```
Comment (svevang) [inline src/app/campaign/store/actions/campaign-action.service.ts:37]: Yes!!

File: src/app/campaign/inventory/goal-form.component.spec.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ class ParentFormComponent {
   @ViewChild('childForm', { static: true }) childForm: GoalFormComponent;
 
   goalForm = this.fb.group({
+    allocationPriority: ['', Validators.min(1)],
```
Comment (svevang) [inline src/app/campaign/inventory/goal-form.component.spec.ts:24]: Min is the max (priority)!
+1

File: src/app/campaign/flight/flight-form.component.html (hunk 1 above)
Comment (svevang) [reply src/app/campaign/flight/flight-form.component.html:50]: Feeder ids are really important and used in a lot of places.  Having a spot to `ctrl-c` and snag them is always nice.

**Review by svevang — APPROVED** [review]
Looks good to me!

**Pull request merged**